### PR TITLE
Sys 4010 make autogrowth mechanism configurable

### PR DESCRIPTION
**Pull request by micaelffrancoAV**
## Proposed changes

- Mechanism to enable or disable growth.
 
## Type of change/Merge

🚨What type of change is this PR? </br>
_Put an `x` in the boxes that apply_

- [ ] Release 
  - [ ] Increase versions 
  - [ ] Baseline tests passed  
  - Release type:
    - [ ] Major release 
    - [ ] Minor release 
    - [ ] Patch release 

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [ ] You describe the purpose of the PR, e.g.:
  - What does it do?
  - Highlight what important points reviewers should know about;
  - Indicates if there is something left for follow-up PRs.
- [ ] Documentation updated
- [ ] Business logic tested successfully
- [ ] Verify First, Write Last: In Substrate development, it is important that you always ensure preconditions are met and return errors at the beginning. After these checks have completed, then you may begin the function's computation.

### CODE REVIEW

**Review by ivan-cholakov — APPROVED** [review]

File: pallets/parachain-staking/src/lib.rs [hunk 1]
```diff
@@ -245,6 +245,8 @@ pub mod pallet {
         type AccountToBytesConvert: pallet_avn::AccountToBytesConverter<Self::AccountId>;
 
         type BridgeInterface: pallet_avn::BridgeInterface;
+
+        type GrowthEnabled: Get<bool>;
```
Comment (nahuseyoum) [inline pallets/parachain-staking/src/lib.rs:249]: if we are setting it in storage, I guess we can remove the config trait?

**Review by nahuseyoum — COMMENTED** [review]
The way its setup now, we need to have a storage migration script to set this value to true so it doesn't break for Aventus so we can either :  
 - include a migration script or 
 - invert the logic and use `growth_disabled`. This way, by default, Growth will be triggered and if a client wants to disable it, they can using the SUDO extrinsic.

File: pallets/parachain-staking/src/lib.rs (hunk 1 above)
Comment (nahuseyoum) [inline pallets/parachain-staking/src/lib.rs:249]:
```suggestion
        #[pallet::constant]
        type GrowthEnabled: Get<bool>;
```

File: pallets/parachain-staking/src/lib.rs [hunk 2]
```diff
@@ -687,6 +689,7 @@ pub mod pallet {
         pub delay: EraIndex,
         pub min_collator_stake: BalanceOf<T>,
         pub min_total_nominator_stake: BalanceOf<T>,
+        pub growth_enabled: bool,
```
Comment (nahuseyoum) [inline pallets/parachain-staking/src/lib.rs:692]: is this needed?

File: pallets/parachain-staking/src/lib.rs [hunk 3]
```diff
@@ -697,6 +700,7 @@ pub mod pallet {
                 delay: Default::default(),
                 min_collator_stake: Default::default(),
                 min_total_nominator_stake: Default::default(),
+                growth_enabled: T::GrowthEnabled::get(),
```
Comment (nahuseyoum) [inline pallets/parachain-staking/src/lib.rs:703]: ditto

File: pallets/parachain-staking/src/tests/mock.rs [hunk 4]
```diff
@@ -566,6 +583,7 @@ impl ExtBuilder {
             delay: 2,
             min_collator_stake: self.min_collator_stake,
             min_total_nominator_stake: self.min_total_nominator_stake,
+            growth_enabled: true,
```
Comment (nahuseyoum) [inline pallets/parachain-staking/src/tests/mock.rs:586]: ditto

File: pallets/parachain-staking/src/tests/test_growth.rs [hunk 5]
```diff
@@ -200,6 +200,71 @@ fn growth_period_indices_updated_correctly() {
...
+                era_index = roll_one_growth_period(era_index);
+
+                let growth_period_info = ParachainStaking::growth_period_info();
+                // ensure indice is not updated
+                assert_eq!(initial_growth_period_index, growth_period_info.start_era_index);
+            });
+    }
+
+    #[test]
+    fn growth_info_is_not_updated() {
```
Comment (nahuseyoum) [inline pallets/parachain-staking/src/tests/test_growth.rs:232]: :+1:

File: pallets/token-manager/src/mock.rs [hunk 6]
```diff
@@ -439,6 +441,7 @@ impl ExtBuilder {
             delay: 2,
             min_collator_stake: 10,
             min_total_nominator_stake: 5,
+            growth_enabled: true,
```
Comment (nahuseyoum) [inline pallets/token-manager/src/mock.rs:444]: and here

File: pallets/validators-manager/src/tests/mock.rs [hunk 7]
```diff
@@ -502,6 +504,7 @@ impl ExtBuilder {
             delay: 2,
             min_collator_stake: 10,
             min_total_nominator_stake: 5,
+            growth_enabled: true,
```
Comment (nahuseyoum) [inline pallets/validators-manager/src/tests/mock.rs:507]: here too

**Review by nahuseyoum — APPROVED** [review]

**Pull request merged**